qa/tasks/cephfs/test_client_limits.py

```diff
@@ -219,6 +220,55 @@ def test_client_release_bug(self):
         self.fs.mds_asok(['session', 'evict', "%s" % mount_a_client_id])
         rproc.wait()
 
+    def test_client_blocklisted_oldest_tid(self):
+        """
+        that a client is blocklisted when its encoded session metadata exceeds the
+        configured threshold (due to ever growing `completed_requests` caused due
+        to an unidentified bug (in the client or the MDS)).
+        """
+
+        # num of requests client issues
+        max_requests = 10000
+
+        # The debug hook to inject the failure only exists in the fuse client
+        if not isinstance(self.mount_a, FuseMount):
+            self.skipTest("Require FUSE client to inject client release failure")
+
+        self.config_set('client', 'client inject fixed oldest tid', 'true')
+        self.mount_a.teardown()
+        self.mount_a.mount_wait()
+
+        self.config_set('mds', 'mds_max_completed_requests', max_requests);
+
+        # Create lots of files
+        self.mount_a.create_n_files("testdir/file1", max_requests + 100)
+
+        # Create a few files synchronously. This makes sure previous requests are completed
+        self.mount_a.create_n_files("testdir/file2", 5, True)
+
+        # Wait for the health warnings. Assume mds can handle 10 request per second at least
+        self.wait_for_health("MDS_CLIENT_OLDEST_TID", max_requests // 10, check_in_detail=str(self.mount_a.client_id))
```

Comment:
@vshankar Please check `check_in_detail` and remove it if the dependent commit seems not being backport yet in `pacific`.

Reply:
I'll pull in the dependent commits and udpate.

Reply:
All 3 dependent commits are already in pacific:

- 3784d9bfb4541de39d803eaed88af7796de87e83
- 5d82bcb21a2ed543218b215d83faea3f8ae7a050
- f1a3ae31c056cccd9bfff7421865781b4ac04bd3